docs/en/03_Upgrading/01_Upgrading_project.md

```diff
@@ -0,0 +1,1188 @@
+Changes not saved; Run with --write to commit to disk
+```
+
+### Manually update deprecated API references
+
+SilverStripe 4 introduces many API changes. To update deprecated API references manually, you have to go through each one of your project files.
+Read the changelogs for [4.0.0](/changelogs/4.0.0/), [4.1.0](/changelogs/4.1.0/) and [4.2.0](/changelogs/4.2.0/)
+for a comprehensive overview.
+
+### Finalising the deprecated API update
+At this stage, your site should be using only SilverStripe 4 API logic.
+
+You still have some minor clean up tasks and configuration tweaks to apply, but you're almost done.
+
+This is a good point to commit your changes to your source control system before moving on to the next step.
+
+## Step 6 - Update your entry point {#step6}
+The location of SilverStripe's _entry file_ has changed. Your project and server environment will need
+to adjust the path to this file from `framework/main.php` to `index.php`.
+
+### Update your `index.php` file
+You can get a copy of the SilverStripe 4 `index.php` file at
+`vendor/silverstripe/recipe-core/public/index.php`.
+
+If you've created your own `index.php` or modified version of `main.php`,
+you'll need to reconcile those changes with the `index.php` file you got from `recipe-core`.
+Otherwise, just use the generic `index.php` file `recipe-core` provides.
+
+Copy your new `index.php` to your project's web root. Unlike SilverStripe 3, `index.php` must be present in your web root.
+
+### Update your server configuration
+If you're using a `.htaccess` file or `web.config` file to handle your server configuration, you can get the generic SilverStripe 4 version of those file from
+`vendor/silverstripe/recipe-core/public`.
+
+Just like `index.php`, if you've modified your server configuration file from the one that shipped with SilverStripe 3, you'll need to reconcile your changes into the version retrieve from `recipe-core`.
+
+[Refer to the installation instruction for your platform](/getting_started/installation/) if your server configuration is not managed via a `.htaccess` or `web.config` file.
+
+### Finalising the entry point upgrade
+
+At this stage, you could in theory run your project in SilverStripe 4.
+
+This is a good point to commit your changes to your source control system before moving on to the next step.
+
+## Step 7 - Update project structure (optional) {#step7}
+SilverStripe 4 introduces a new recommended project structure ([details](/changelogs/4.2.0#app-folder-name)).
+Adopting the recommended project structure is optional, but will become mandatory in SilverStripe 5.
+
+[Skip to Step 8](#step8)
+
+### Automatically switch to the new structure with the `reorganise` command
+The reorganise command can automatically update your project to use the new recommended structure.
+
+It will search your code and find any occurrence of `mysite`. It won't replace those occurrence with `app` however.
```

Comment:
```suggestion
It will search your code and find any occurrence of `mysite`. It won't replace those occurrences with `app` however.
```

As a uneducated user - what does this do then? Does it not replace it _unless I specify `--write`_? If so this bit reads a bit differently than others.